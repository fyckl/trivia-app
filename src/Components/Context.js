import React, { useContext, useState, useEffect } from 'react'

const Context = React.createContext()

function ContextProvider(props) {

    const [startQuiz, setStartQuiz] = useState(false)
    const [arrQuestions, setArrQuestions] = useState([])
    const [playAgain, setPlayAgain] = useState(false)
    const [correctAnswers, setCorrectAnswers] = useState(0)
    const [score, setScore] = useState(0)
    const [shuffleAnswers, setShuffleAnswers] = useState(false)

    function quizStart(){
      setShuffleAnswers(true)
      setStartQuiz(true)
    }

    function checkAnswers(){
      setShuffleAnswers(false)
      setPlayAgain(true)
    }

    useEffect(() => {
      setScore(prevState => prevState + correctAnswers)
    }, [correctAnswers])

    function reset(){
      setArrQuestions([])
      getQuestions()
      setPlayAgain(false)
      setCorrectAnswers(0)
      setShuffleAnswers(true)
    }

    function getQuestions(){
      fetch('https://opentdb.com/api.php?amount=5&difficulty=easy&type=multiple&encode=base64')
      .then(res=> res.json())
      .then(data=> setArrQuestions(data.results))
    }

    
    // console.log(arrQuestions)
    useEffect(() => {
      getQuestions()
    }, [])

    return(
        <Context.Provider 
            value={{
                quizStart, 
                startQuiz, 
                arrQuestions,
                setCorrectAnswers,
                checkAnswers,
                playAgain,
                correctAnswers,
                reset,
                score,
                shuffleAnswers
        }}>
            {props.children}
        </Context.Provider>
    )
}

export {ContextProvider, Context}