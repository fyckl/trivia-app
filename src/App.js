import React, { useState, useEffect, useContext } from "react";
import Start from "./Components/Start";
import Questions from "./Components/Questions";
import Answers from "./Components/Answer";
import { Context } from "./Components/Context"
function App() {
  const {quizStart, startQuiz} = useContext(Context)

  return (
    <div className="main-container">
      {startQuiz ? <Questions /> : <Start start={quizStart}/>}
    </div>
  );

}

export default App;
