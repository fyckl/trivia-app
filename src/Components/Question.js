import React, { useState, useEffect, useContext } from 'react'
import Answer from './Answer'
import { Context, useAppContext } from "./Context"
import { Buffer } from 'buffer'
import useShuffleArr from '../Hooks/useShuffleArr'

function Question({question, incorrectAnswers, correctAnswer}) {
  
  const arrAnswers = incorrectAnswers.concat(correctAnswer)
  const [shuffledAnswers, setShuffledAnswers] = useState([])
  const { isShuffleAnswers } = useAppContext()
  const shuffleArr = useShuffleArr(arrAnswers)

  useEffect(() => {
    if(isShuffleAnswers){
      setShuffledAnswers(shuffleArr)
    }
  }, [isShuffleAnswers])

  const answers = shuffledAnswers.map((item) => {
      return <Answer key={item} answers={item} correctAnswer={correctAnswer}/>
  })
  const decodedQuestion = Buffer.from(question, 'base64').toString()
  return (
    <div className='question-container' data-testid="container">
        <h3 className='question'>{decodedQuestion}</h3>
        {answers}
        <hr className='hr'/>
    </div>
  )
}

export default Question