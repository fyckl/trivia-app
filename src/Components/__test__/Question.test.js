import React from "react"
import Question from "../Question"
import * as Context  from "../Context"
import App from "../../App"
import { fireEvent, cleanup, screen, render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { Enzyme, shallow, configure, mount } from 'enzyme'
import { Buffer } from 'buffer'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test("renders correctly to match snapshot", () => {
    const contextValues = {
        isShuffleAnswers: true
    }

    jest
      .spyOn(Context, 'useAppContext')
      .mockImplementation(() => contextValues);

    const wrapper = shallow(<Question incorrectAnswers={[1,2,3]} correctAnswer={[4]} question=""/>)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find(Question)).toBeTruthy();

   
})