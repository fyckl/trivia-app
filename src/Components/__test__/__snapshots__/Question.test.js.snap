// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly to match snapshot 1`] = `"<div class=\\"question-container\\" data-testid=\\"container\\"><h3 class=\\"question\\"></h3><hr class=\\"hr\\"/></div>"`;
